architecture/2024-08-controller-component-threat-model.md

```diff
@@ -0,0 +1,241 @@
+2. **Avoid Custom Parsers**: By not using custom or third-party parsers, Radius avoids the additional risk and complexity associated with ensuring the security of these parsers.
+3. **Fuzz Testing**: If a custom parser were to be used, it would be essential to implement fuzz testing. Fuzz testing involves providing invalid, unexpected, or random data inputs to the parser to identify potential vulnerabilities. For more information on fuzz testing, refer to [OWASP Fuzzing](https://owasp.org/www-community/Fuzzing).
+
+#### Users with access to the webhook server modifying the behavior of the webhook server
+
+Risk: Could an insider with access to the webhook server modify its behavior to approve malicious requests?
+
+Mitigation: Implement strict access controls, audit logs, and monitoring for changes to the webhook server configuration and code.
+
+#### Unauthorized access to the Kubernetes API Server
+
+- **Mitigation**: Implement strong authentication and authorization mechanisms, such as RBAC (Role-Based Access Control) and network policies to restrict access.
+
+#### Security vulnerabilities in the controller code
+
+- **Mitigation**: Conduct regular security audits and code reviews, and follow secure coding practices to minimize vulnerabilities.
+
+## Open Questions
+
+<!--
+List any unresolved questions or uncertainties about the threat model. Use this section to gather feedback from experts or team members and to track decisions made during the review process.
+-->
+
+## Action Items
+
+1. Check if TLS is enabled for every component to ensure secure communication. Make changes to the necessary components if required.
+2. Ensure that all communication uses mTLS (Mutual TLS) to authenticate both the client and server, providing an additional layer of security. Verify that mTLS is correctly configured for all components and endpoints. Make changes to the necessary components if required.
+3. Check if RBAC with Least Privilege is configured for every component to ensure that each component has only the permissions it needs to function. Make changes to the necessary components if required.
+4. Define and implement necessary Network Policies to ensure that communication is accepted only from expected and authorized components. Regularly review and update these policies to maintain security.
+5. Separate and firewall the etcd cluster to ensure the safety of the datastore. Implement network segmentation to isolate the etcd cluster from other components. Configure firewall rules to restrict access to the etcd cluster, allowing only authorized components and administrators to communicate with it. Regularly review and update firewall rules and network policies to maintain security.
+6. Containers should run as a non-root user wherever possible to minimize the risks. Check if we can run any of the Radius containers as non-root. Do the necessary updates.
+
+## Review Notes
```

Comment:
What if user registered a malicious recipe or just versions of resource that have known vulnerabilities? the audit logs could capture who/ when of the event, but would the scenario fall under radius related security?